src/opera/parser/tosca/csar.py

```diff
@@ -1,31 +1,286 @@
 import shutil
+from abc import ABC, abstractmethod
+from pathlib import Path, PurePath
+from tempfile import mktemp
+from typing import List, IO, Optional
+from zipfile import ZipFile
+
 import yaml
 
-from pathlib import Path
-from tempfile import TemporaryDirectory, mktemp
-from zipfile import ZipFile
+from opera.error import ParseError, OperaError
+from opera.utils import determine_archive_format
+
+
+class CsarMeta:
+    SUPPORTED_META_FILE_VERSIONS = {"1.1"}
+    SUPPORTED_CSAR_VERSIONS = {"1.1"}
+    METADATA_PATH = "TOSCA-Metadata/TOSCA.meta"
+
+    KEY_NAME = "Name"
+    KEY_CONTENT_TYPE = "Content-Type"
+    KEY_TOSCA_META_FILE_VERSION = "TOSCA-Meta-File-Version"
+    KEY_CSAR_VERSION = "CSAR-Version"
+    KEY_CREATED_BY = "Created-By"
+    KEY_ENTRY_DEFINITIONS = "Entry-Definitions"
+
+    REQUIRED_KEYS = {KEY_TOSCA_META_FILE_VERSION, KEY_CSAR_VERSION, KEY_CREATED_BY}
+
+    def __init__(self, name: str, content_type: str, tosca_meta_file_version: str,
+                 csar_version: str, created_by: str, entry_definitions: str):
+        self.name = name
+        self.content_type = content_type
+        self.tosca_meta_file_version = tosca_meta_file_version
+        self.csar_version = csar_version
+        self.created_by = created_by
+        self.entry_definitions = entry_definitions
+
+    @classmethod
+    def parse(cls, contents: str) -> "CsarMeta":
+        # FIXME: the meta format is not yaml, but it's close enough to parse a simple file
```

Comment:
How can this be fixed in the future?

Reply:
By writing a meta file parser. The format is a bit underspecified though.

Reply:
Aha ok, we will se if the meta file will be more spcified in TOSCA Simple Profile in YAML `v2.0`.